modules/oe_theme_inpage_navigation/tests/src/Kernel/Plugin/Condition/InPageNavigationStateTest.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_inpage_navigation\Kernel\Plugin\Condition;
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Plugin\Context\EntityContext;
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+use Drupal\oe_theme_inpage_navigation\InPageNavigationHelper;
+use Drupal\Tests\oe_theme\Kernel\AbstractKernelTestBase;
+
+/**
+ * Test InPageNavigationState plugin.
+ */
+class InPageNavigationStateTest extends AbstractKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'emr',
+    'emr_node',
+    'node',
+    'field',
+    'entity_reference_revisions',
+    'oe_theme_inpage_navigation',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('entity_meta');
+    $this->installEntitySchema('entity_meta_relation');
+    $this->installSchema('node', ['node_access']);
+
+    $this->installConfig([
+      'emr',
+      'emr_node',
+      'oe_theme_inpage_navigation',
+    ]);
+  }
+
+  /**
+   * Tests the inpage navigation state condition.
+   *
+   * @param bool|null $inpage_navigation_condition
+   *   Inpage navigation state condition.
+   * @param bool|null $inpage_navigation_condition_negate
+   *   Is negate.
+   * @param array|null $inpage_navigation_node_state
+   *   Inpage navigation state in node.
+   * @param array $expected
+   *   Array with expected values of 'summary' and 'result'.
+   *
+   * @dataProvider providerTestCondition
+   */
+  public function testCondition(?bool $inpage_navigation_condition, ?bool $inpage_navigation_condition_negate, ?array $inpage_navigation_node_state, array $expected): void {
+    $manager = \Drupal::service('plugin.manager.condition');
+    $condition = $manager->createInstance('oe_theme_inpage_navigation_state');
+    $condition->setConfiguration(['inpage_navigation_state' => $inpage_navigation_condition, 'negate' => $inpage_navigation_condition_negate]);
+
+    if ($inpage_navigation_node_state) {
+      $condition->setContextMapping([
+        'node' => 'node',
+      ]);
+      NodeType::create(['type' => 'example', 'name' => 'Example'])->save();
```

Comment:
Provided code is correct but please consider using build/core/modules/node/tests/src/Traits/ContentTypeCreationTrait.php and build/core/modules/node/tests/src/Traits/NodeCreationTrait.php:
```
    $this->drupalCreateContentType([
      'type' => 'page',
      'name' => 'Page',
    ]);
$node = $this->drupalCreateNode();
```
because it is commonly used way to create test nodes in contrib modules.